openedx/core/djangoapps/plugins/README.rst

```diff
@@ -184,6 +184,19 @@ class::
                         PluginSignals.SENDER_PATH: u'full_path_to_sender_app.ModelZ',
                     }],
                 }
+            },
+
+            # Configuration setting for Plugin Contexts for this app.
+            PluginContexts.CONFIG: {
+
+                # Configure the Plugin Signals for each Project Type, as needed.
+                ProjectType.LMS: {
+
+                    # Key is the view that the app wishes to add context to and the value
+                    # is the function within the app that will return additional context
+                    # when called with the original context
+                    PluginContexts.VIEWS.STUDENT_DASHBOARD: u'my_app.context_api.get_dashboard_context'
```

Comment:
**Question:** While the backend plugin can add fields to the View's context, how do you plan to extend the View templates to make use of the pluggable contexts?

Reply:
For the time being, I was planning on relying on theming for the template side of things. I think the longer term option (the time between after this proposal and before MFEs take over) would be to break each template into a series of sections, which can them have template code added to them dynamically, similar to what we're doing with the context here. The dashboard side panel could check all plugins for additional template code via namespace and append them. Something like 
```
u'template_config': {
    u'lms.djangoapp': {
        'student_dashboard.side_panel': u'my_app.template_api.get_student_dashboard_side_panel_template'
    }
}
```

We would have to make the concious decision of what to make extensible and what to become a named section that can be edited. I imagine we're going to have to make these decisions when we handle plugable MFEs as well, but maybe there's a simpler option there that I'm missing.